### PR TITLE
chore: eslint 9 import plugin update

### PR DESCRIPTION
**Pull request by algreasley**
as [Alpaca PR](https://github.com/weareadaptive/alpaca-reference-ui/pull/290), just keeping up to date - removing `@eslint/compat` plus all possible "overrides")

### CODE REVIEW

File: packages/client/package.json [hunk 1]
```diff
@@ -121,33 +119,33 @@
...
-    "concurrently": "8.2.2",
+    "@vitest/browser": "2.1.2",
+    "@vitest/eslint-plugin": "1.1.5",
+    "concurrently": "9.0.1",
     "cross-env": "^6.0.3",
     "csstype": "^3.0.2",
-    "eslint": "9.9.0",
-    "eslint-plugin-import": "2.29.1",
+    "eslint": "9.11.1",
+    "eslint-plugin-import": "2.31.0",
```
Comment (algreasley) [inline packages/client/package.json:128]: this is the important bump

File: packages/client/src/client/App/Credit/CreditRfqForm/CreditInstrumentSearch.tsx [hunk 2]
```diff
@@ -60,9 +60,9 @@ export const [useSelectedInstrument] = bind(
     withLatestFrom(creditInstrumentsByCusip$),
     map(([instrumentId, creditInstrumentsByCusip]) =>
       instrumentId !== null
-        ? Object.values(creditInstrumentsByCusip).find(
+        ? (Object.values(creditInstrumentsByCusip).find(
```
Comment (algreasley) [inline packages/client/src/client/App/Credit/CreditRfqForm/CreditInstrumentSearch.tsx:63]: only 1 mod in this project 😅 
surprising there was any change, tho, tbh as I only bumped 1 patch level!

File: packages/client/package.json [hunk 3]
```diff
@@ -95,13 +95,11 @@
     "styled-components": "5.3.9"
   },
   "overrides": {
-    "eslint": "9.9.0",
     "next": "14.2.13"
   },
   "devDependencies": {
     "@adaptive/hydra-web-codegen": "4.57.0",
-    "@eslint/js": "9.9.0",
-    "@eslint/compat": "1.1.1",
```
Comment (algreasley) [inline packages/client/package.json:104]: compat no longer necessary with shift to latest import plugin 🙂

File: packages/client/package.json [hunk 4]
```diff
@@ -95,13 +95,11 @@
     "styled-components": "5.3.9"
   },
   "overrides": {
-    "eslint": "9.9.0",
```
Comment (algreasley) [inline packages/client/package.json:98]: need for eslint override gone with ESLint 9-compatible import plugin

Comment (algreasley) [reply packages/client/package.json:98]: next still required due to OF WS libs incorrect categorisation of deps / numerous vulnerabilities due to the old version of next that they specify 🤦🏻

File: packages/client/eslint.config.js [hunk 5]
```diff
@@ -44,10 +43,10 @@ export default [
       // https://github.com/facebook/react/tree/main/packages/eslint-plugin-react-hooks
       // compat with ESLint v9, see https://github.com/facebook/react/issues/28313
       // recent "compatibility" PR does not address flat config at all, so still on "hacked" eslint config
+      // also need to use "rc" version - been rc for 5mths! - otherwise peer deps complaints
```
Comment (algreasley) [inline packages/client/eslint.config.js:46]: I forgot about this when I migrated Alpaca - at some point, perhaps, they will release 5.1.0, but as the comment says it has been RC for a while now ..

**Review by SHolleworth — APPROVED** [review]

**Pull request merged**